File: python/Day24/more_ll.py
def removeDuplicates(self,head):
  #Write your code here
  if head is None or head.next is None:
      return head
  
  cur = head.next
  prev = head
  while cur is not None:
      if cur.data == prev.data:
	  prev.next = cur.next
	  
	  temp = cur
	  cur = cur.next
	  temp.next = None
      else:
	  prev = cur
	  cur = cur.next
  
  return head

File: cpp/Day24/more_ll.cpp
Node* removeDuplicates(Node *head)
{
  //Write your code here
  if(head == NULL || head->next == NULL){
    return head;
  }

  Node *temp_prev = head;
  Node *temp_cur = head->next;
  while(temp_cur != NULL){
    int cur_data = temp_cur->data;
    int prev_data = temp_prev->data;

    if(cur_data == prev_data){
      temp_prev->next = temp_cur->next;

      //Deleting the node
      Node* temp = temp_cur;
      temp_cur = temp_cur->next;
      free(temp);

    }else{
      temp_prev = temp_cur;
      temp_cur = temp_cur->next;
    }
  }

  return head;
}
